### DISCUSSION TITLE
Failed to find a new element for particle

### DISCUSSION THREAD

**Discussion by QZheng82** [Anura3D/Anura3D_OpenSource#109] [Q&A] [answered]
Hello. I made a 3D slope with anti-slip piles to simulate someone's sliding surface. The material parameters are guaranteed to be fine, but it always appears : Failed to find a new element for particle xx in time step xx at position xx xx xx.
![VM~O11L $0Q~FI7(84B~M M](https://github.com/Anura3D/Anura3D_OpenSource/assets/119284513/db8654a1-7280-4285-acee-c1029bbb4ac7)
![ND}~1)7KXE 8PSPW%0CRVYV](https://github.com/Anura3D/Anura3D_OpenSource/assets/119284513/896546f0-856d-4f3a-877f-ed65b1f7c61e)
![04P1`_BKJ5K8FQ2TRMQA(5C](https://github.com/Anura3D/Anura3D_OpenSource/assets/119284513/6dd5f87f-82bf-4377-a298-259659501408)
![X) %} X4YCSM(97%BYH2{A9](https://github.com/Anura3D/Anura3D_OpenSource/assets/119284513/0923a5f3-311f-4c28-927c-ca3f79f31ad0)

 I would like to know if I need to set boundary conditions internally or if it is some other problem?Thank you!
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/119284513/b72c8977-6f7d-4241-b4b8-5d8b7077a066)

**Comment by Francisliii** [comment]
Hi,

I had the same issue in one of my model. I don't know you calculations data and other parameters, but you may want to try finer mesh in areas where the error appears. 

Francis

**Reply by QZheng82** [reply]
After the grid is adjusted more accurately, there will still be errors, could it be that the calculation parameters are not set properly?

**Comment by Karl131127** [comment]
There was a same error in my model. I'd like to know that how you guys solve this problem. Cuz even I had made  better mesh, the error still occurred.

**Reply by QZheng82** [reply]
No suitable solution has been found yet.

**Reply by Karl131127** [reply]
Hi, have you been able to solve this problem? Or some experience, like something we need to pay attention?

**Comment by murphy-john** [comment, marked as answer]
"Failed to find a new element for particle xx in time step xx at position xx xx xx."  usually means one of the Material Points has left the mesh. You can use the [debug feature](https://docs.microsoft.com/en-us/visualstudio/debugger/debugging-absolute-beginners?view=vs-2022&tabs=csharp) in Visual Studio to locate the Material Point which has left. 

There are a few ways this issue can happen. First, when the boundary conditions is not applied the a Material Point is allowed to leave the mesh. Second, there is a numerical instability. To fix this double check your stress initialization. You can also try reducing the Courant number.
